install/plugin-updater.php

```diff
@@ -151,6 +154,20 @@ public function get_repo_api_data() {
 			$this->set_version_info_cache( $version_info );
 		}
 
+		// Added by Polylang.
+		foreach ( array( 'url', 'package', 'new_version', 'tested', 'requires', 'requires_php' ) as $prop ) {
+			if ( ! isset( $version_info->$prop ) ) {
+				$version_info->$prop = '';
+			}
+		}
+
+		foreach ( array( 'icons', 'banners' ) as $prop ) {
+			if ( ! isset( $version_info->$prop ) ) {
+				$version_info->$prop = new stdClass();
+			}
+		}
```

Comment:
What do you think of
```suggestion
		$default = array(
			'url'          => '',
			'package       => '',
			'new_version'  => '',
			'tested'       => '',
			'requires'     => '',
			'requires_php' => '',
			'icons'        => new stdClass(),
			'banners'      => new stdClass(),
		);
		
		$version_info = (object) array_merge( $defaults, (array) $version_info );
```

Reply:
That's indeed a solution I thought about.